src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -1266,14 +1266,12 @@ export default class EthereumApi implements Api {
   @assertArgLength(1)
   async eth_getBlockTransactionCountByHash(hash: DATA) {
     const { blocks } = this.#blockchain;
-    const blockNum = await blocks.getNumberFromHash(hash);
-    if (!blockNum) return null;
-
-    const rawBlock = await blocks.getRawByBlockNumber(Quantity.from(blockNum));
-    if (!rawBlock) return null;
-
-    const [, rawTransactions] = decode<GanacheRawBlock>(rawBlock);
-    return Quantity.from(rawTransactions.length);
+    const block = await blocks
```

Comment:
Looks like the bug is really in `getNumberFromHash`, if we fix that I suspect the fix for https://github.com/trufflesuite/ganache/pull/3692 will become slightly simpler.

Do you want to try to tackle fixing `getNumberFromHash` to fetch from the fallback?

Reply:
getByHash does the right thing (calling eth_getBlockByHash on the fork) and getNumberFromHash end up being just a helper function we can even declare it private in the future.